File: go/top-k-frequent-elements.go
func topKFrequent(nums []int, k int) []int {
	m := map[int]int{}
	for _, v := range nums {
		m[v]++
	}

	type data struct {
		num, cnt int
	}

	st := []data{}
	for i, v := range m {
		st = append(st, data{i, v})
	}

	sort.SliceStable(st, func(i, j int) bool {
		return st[i].cnt > st[j].cnt
	})

	res := make([]int, k)
	for i := 0; i < k; i++ {
		res[i] = st[i].num
	}

	return res
}

File: cpp/top-k-frequent-elements.cpp
class Solution {
public:
    struct Data {
        int num, cnt;
        bool operator < (const Data &o) const {
            return cnt < o.cnt;
        }
    };
    
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> m;
        for (int x: nums) m[x]++;
        
        priority_queue<Data> st;
        for (auto x: m) {
            st.push({ x.first, x.second });
        }
        
        vector<int> res;
        while (k--) {
            Data u = st.top();
            res.push_back(u.num);
            st.pop();
        }
        
        return res;
    }
};